modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/EnterpriseGeoIpDownloader.java

```diff
@@ -470,31 +481,20 @@ private void runOnDemand() {
             logger.debug("Not running downloader on demand because task is cancelled or completed");
             return;
         }
-        // Here we do want to wait for the current run (if any) to finish. Since a new cluster state might have arrived while the current
-        // run was running, we want to ensure that the new cluster state update isn't lost, so we wait and run afterwards.
-        logger.trace("Waiting to run downloader on demand");
-        boolean acquired = false;
+        // Capture the current queue size, so that if another run is requested while we're running, we'll know at the end of this method
+        // whether we need to run again.
+        final int currentQueueSize = queuedRuns.get();
```

Comment:
nit, up to you, mainly thinking out loud

imo while reading code it's not instantly clear at this point that this path is guaranteed by `queuedRuns` to not be running concurrently (which would cause issues), i had to double-check this when reviewing

may i suggest

```suggestion
        // whether we need to run again.
        // n.b. this function is protected by queuedRuns to not run concurrently, so don't worry about concurrency/races.
        final int currentQueueSize = queuedRuns.get();
```